From 001dff36b223612e08e9c96bae08925b2f5b2e71 Mon Sep 17 00:00:00 2001
From: Ihor Mashevskyi <i.mashevskyi@dev-branch.com>
Date: Mon, 7 Jul 2025 15:36:29 +0300
Subject: [PATCH] Issue #3516018: The title for layout_builder blocks can't be
 edited

---
 frontend_editing.module | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/frontend_editing.module b/frontend_editing.module
index 10e5766..cf1aea6 100644
--- a/frontend_editing.module
+++ b/frontend_editing.module
@@ -15,6 +15,7 @@ use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
 use Drupal\Core\Entity\EntityForm;
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\RevisionableInterface;
+use Drupal\Core\Entity\TranslatableInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Form\SubformState;
 use Drupal\Core\Render\Element;
@@ -1093,7 +1094,7 @@ function _frontend_editing_update_parent_layout_builder(array $form, FormStateIn
   }
 
   // Get correct translation.
-  if ($parent_entity->hasTranslation($langcode)) {
+  if ($parent_entity instanceof TranslatableInterface && $parent_entity->hasTranslation($langcode)) {
     $parent_entity = $parent_entity->getTranslation($langcode);
   }
 
-- 
GitLab

