diff --git a/trash.module b/trash.module
index 612e99a..2985a73 100644
--- a/trash.module
+++ b/trash.module
@@ -321,7 +321,10 @@ function trash_form_alter(&$form, FormStateInterface $form_state, $form_id) {
   ];
 
   // Use different messages based on the user's access level.
-  if (\Drupal::currentUser()->hasPermission('restore ' . $entity_type->id() . ' entities')) {
+  if ($is_entity_delete_form && $entity instanceof TranslatableInterface && $entity->isTranslatable() && !$entity->isDefaultTranslation()) {
+    $entity_delete_label = t('Deleting this @label will delete it from the site and this action cannot be undone.', $params);
+  }
+  elseif (\Drupal::currentUser()->hasPermission('restore ' . $entity_type->id() . ' entities')) {
     $trash_settings = \Drupal::configFactory()->get('trash.settings');
     if ($trash_settings->get('auto_purge.enabled')) {
       $timestamp = strtotime(sprintf('+%s', $trash_settings->get('auto_purge.after')));
@@ -344,41 +347,6 @@ function trash_form_alter(&$form, FormStateInterface $form_state, $form_id) {
     $entity_multiple_delete_label = t('Deleting these @label will move them to the trash.', $params);
   }
 
-  // Prevent deleting individual translations.
-  // @todo Remove this after https://www.drupal.org/i/3376216 is fixed.
-  if ($is_entity_delete_form && $entity instanceof TranslatableInterface && $entity->isTranslatable() && !$entity->isDefaultTranslation()) {
-    $entity_delete_label = t('Deleting a translation of a @label is currently not supported by Trash. Unpublish the translation instead.', $params);
-    $form['actions']['submit']['#access'] = FALSE;
-  }
-  elseif ($is_entity_multiple_delete_form && $entity_type->isTranslatable()) {
-    $storage = \Drupal::entityTypeManager()->getStorage($entity_type->id());
-
-    $can_delete = TRUE;
-    foreach (array_keys($form['entities']['#items'] ?? []) as $item) {
-      [$id, $langcode] = explode(':', $item, 2);
-
-      // All entities have been loaded already in the static cache by the
-      // delete multiple form, so it's ok to single-load them again.
-      $entity = $storage->load($id);
-      assert($entity instanceof TranslatableInterface);
-
-      // Deleting the default translation is considered the same as deleting
-      // the entire entity. When all translations are selected, only the
-      // default langcode will show up in the selections.
-      if (!$entity->getTranslation($langcode)->isDefaultTranslation()) {
-        // If any of the selected translations are not the default translation
-        // of the entity, the multiple deletion can not proceed.
-        $can_delete = FALSE;
-        break;
-      }
-    }
-
-    if (!$can_delete) {
-      $entity_multiple_delete_label = t('Deleting translations of @label is currently not supported by Trash. Unpublish the translations instead.', $params);
-      $form['actions']['submit']['#access'] = FALSE;
-    }
-  }
-
   $trash_handler = \Drupal::service('trash.manager')->getHandler($entity_type->id());
   assert($trash_handler instanceof TrashHandlerInterface);
   if (isset($form['description']['#markup']) && $form['description']['#markup'] instanceof TranslatableMarkup) {
